Include/pylifecycle.h

```diff
@@ -133,6 +133,12 @@ PyAPI_FUNC(int) _PyOS_URandom(void *buffer, Py_ssize_t size);
 PyAPI_FUNC(int) _PyOS_URandomNonblock(void *buffer, Py_ssize_t size);
 #endif /* !Py_LIMITED_API */
 
+/* Legacy locale support */
+#ifndef Py_LIMITED_API
+PyAPI_FUNC(void) _Py_CoerceLegacyLocale(void);
```

Comment:
If add declarations here they should be removed from `Programs/python.c`.

Reply:
Indeed; I just wonder if there was some reason this was done the way it was done in the first place. I can't see a good reason, particularly...

Reply:
The only reason I did it that way was because they're private APIs intended specifically for the CPython CLI, and that code built OK in CI and on the buildbots.

However, since that turns out to cause problems on Cygwin, switching to the more conventional approach makes sense, and the private declarations can be dropped.

Reply:
Okay, thanks for the clarification.